cub/cub/device/device_select.cuh

```diff
@@ -947,72 +957,220 @@ struct DeviceSelect
             typename ValueInputIteratorT,
             typename KeyOutputIteratorT,
             typename ValueOutputIteratorT,
-            typename NumSelectedIteratorT>
-  CUB_RUNTIME_FUNCTION _CCCL_FORCEINLINE static cudaError_t
-  UniqueByKey(void *d_temp_storage,
-              size_t &temp_storage_bytes,
-              KeyInputIteratorT d_keys_in,
-              ValueInputIteratorT d_values_in,
-              KeyOutputIteratorT d_keys_out,
-              ValueOutputIteratorT d_values_out,
-              NumSelectedIteratorT d_num_selected_out,
-              int num_items,
-              cudaStream_t stream = 0)
+            typename NumSelectedIteratorT,
+            typename NumItemsT,
+            typename EqualityOpT>
+  CUB_RUNTIME_FUNCTION __forceinline__ static //
+    typename ::cuda::std::enable_if< //
+      !::cuda::std::is_convertible<EqualityOpT, cudaStream_t>::value, //
+      cudaError_t>::type
+    UniqueByKey(
+      void* d_temp_storage,
+      size_t& temp_storage_bytes,
+      KeyInputIteratorT d_keys_in,
+      ValueInputIteratorT d_values_in,
+      KeyOutputIteratorT d_keys_out,
+      ValueOutputIteratorT d_values_out,
+      NumSelectedIteratorT d_num_selected_out,
+      NumItemsT num_items,
+      EqualityOpT equality_op,
```

Comment:
suggestion: docs above state that "The `==` equality operator is used to determine whether keys are equivalent". I suggest to change that to "The `equality_op` is used to determine whether keys are equivalent". Also, please, update the code snippet to provide `equality_op`.